wallet/how-to/connect/set-up-sdk/mobile/android.md

```diff
@@ -168,64 +142,105 @@ details on how to set up a Jetpack Compose project to work with the SDK.
 Import the SDK by adding the following line to the top of your project file:
 
 ```kotlin
-import io.metamask.androidsdk.EthereumViewModel
+import io.metamask.androidsdk.Ethereum
 ```
 
 ### 4. Connect your dapp
 
-Connect your dapp to MetaMask by adding the following code to your project file.
-The Ethereum module requires the app context, so you must instantiate it from an activity or a
-module that injects a context.
+You can connect your dapp to MetaMask in one of two ways:
 
-```kotlin title="MainActivity.kt"
-// MainActivity
+1. [Use the `ethereum` provider object directly](#41-use-the-provider-object-directly).
+    We recommend using this method in a pure model layer.
+2. [Use a ViewModel](#42-use-a-viewmodel) that injects the `ethereum` provider object.
+    We recommend using this method at the app level, because it provides a single instance that
+    survives configuration changes and can be shared across all views.
 
-// Obtain EthereumViewModel using viewModels()
-val ethereumViewModel: EthereumViewModel by viewModels()
+#### 4.1. Use the provider object directly
 
-val dapp = Dapp(name: "Droid Dapp", url: "https://droiddapp.com")
+Use the `ethereum` provider object directly to connect your dapp to MetaMask by adding the following
+code to your project file:
 
-// This is the same as calling "eth_requestAccounts"
-ethereumViewModel.connect(dapp) { result ->
-    if (result is RequestError) {
-        Log.e(TAG, "Ethereum connection error: ${result.message}")
-    } else {
-        Logger.d(TAG, "Ethereum connection result: $result")
+```kotlin
+@AndroidEntryPoint
+class SomeModel(private val repository: ApplicationRepository) {
+    val ethereum = Ethereum(repository)
+    
+    val dapp = Dapp("Droid Dapp", "https://droiddapp.com")
+
+    // This is the same as calling eth_requestAccounts
+    ethereum.connect(dapp) { result ->
+        if (result is RequestError) {
+            Log.e(TAG, "Ethereum connection error: ${result.message}")
+        } else {
+            Log.d(TAG, "Ethereum connection result: $result")
+        }
     }
 }
 ```
 
-By default, MetaMask logs three SDK events: `connectionRequest`, `connected`, and `disconnected`.
-This allows MetaMask to monitor any SDK connection issues.
-To disable this, set `ethereumViewModel.enableDebug = false`.
+#### 4.2. Use a ViewModel
 
-### 5. Call provider methods
+To connect your dapp to MetaMask using a ViewModel, first create a ViewModel that injects the
```

Comment:
For the statement... To connect your dapp to MetaMask using a ViewModel, first create a ViewModel that injects the...

You really don't need "first".